cmd/esc/cli/env_clone.go

```diff
@@ -0,0 +1,97 @@
+// Copyright 2023, Pulumi Corporation.
+
+package cli
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/pulumi/esc/cmd/esc/cli/client"
+	"github.com/spf13/cobra"
+)
+
+func newEnvCloneCmd(env *envCommand) *cobra.Command {
+	var (
+		preserveHistory         bool
+		preserveAccess          bool
+		preserveEnvironmentTags bool
+		preserveRevisionTags    bool
+	)
+
+	cmd := &cobra.Command{
+		Use:   "clone [<org-name>/]<project-name>/<environment-name> [<project-name>/]<environment-name>",
```

Comment:
it'll be wordy but it would be nice to be clear which one is the old vs new. maybe `existing-project-name/existing-envionment-name` vs `new-project-name/new-environment-name...`, or `src vs dest` terminology

Reply:
I'll add in src and dest